tasks/simplification.py

```diff
@@ -0,0 +1,227 @@
+                  shapefile=self.output().path, schema=self.schema, table=self.table,
+                  skipfailures='-skipfailures' if self.skipfailures.lower() == SKIPFAILURES_YES else '')
+        shell(cmd)
+
+    def output(self):
+        return LocalTarget(os.path.join(tmp_directory(self.schema, self.table), shp_filename(self.schema, self.table)))
+
+
+class SimplifyShapefile(Task):
+    schema = Parameter()
+    table = Parameter()
+    retainpercentage = Parameter(default=DEFAULT_M_RETAIN_PERCENTAGE)
+    skipfailures = Parameter(default=SKIPFAILURES_NO)
+
+    def requires(self):
+        yield ExportShapefile(schema=self.schema, table=self.table, skipfailures=self.skipfailures)
+
+    def run(self):
+        cmd = 'node --max-old-space-size=8192 `which mapshaper` ' \
+              '{input} snap -simplify {retainpercentage}% keep-shapes -o {output}'.format(
+                input=os.path.join(tmp_directory(self.schema, self.table),
+                                   shp_filename(self.schema, self.table)),
+                retainpercentage=self.retainpercentage,
+                output=self.output().path)
+        shell(cmd)
+
+    def output(self):
+        return LocalTarget(os.path.join(tmp_directory(self.schema, self.table),
+                                        shp_filename(self.schema, self.table, SIMPLIFIED_SUFFIX)))
+
+
+class ImportSimplifiedShapefile(Task):
+    schema = Parameter()
+    table = Parameter()
+    retainpercentage = Parameter(default=DEFAULT_M_RETAIN_PERCENTAGE)
+    skipfailures = Parameter(default=SKIPFAILURES_NO)
+
+    executed = False
+
+    def requires(self):
+        yield SimplifyShapefile(schema=self.schema, table=self.table, retainpercentage=self.retainpercentage,
+                                skipfailures=self.skipfailures)
+
+    def run(self):
+        session = CurrentSession().get()
+
+        session.execute('TRUNCATE TABLE "{schema}".{table} '.format(
+                         schema=self.output().schema, table=self.output().tablename))
+        session.execute("COMMIT")
+
+        cmd = 'PG_USE_COPY=yes ' \
+              'ogr2ogr -f PostgreSQL "PG:dbname=$PGDATABASE active_schema={schema}" ' \
+              '-t_srs "EPSG:4326" -nlt MultiPolygon -nln {table} ' \
+              '-lco OVERWRITE=yes -lco PRECISION=no -lco GEOMETRY_NAME= ' \
+              '-lco SCHEMA={schema} {shp_path} '.format(
+                    schema=self.output().schema,
+                    table=self.output().tablename,
+                    shp_path=os.path.join(tmp_directory(self.schema, self.table),
+                                          shp_filename(self.schema, self.table, SIMPLIFIED_SUFFIX)))
+        shell(cmd)
+        self.executed = True
+
+    def output(self):
+        return PostgresTarget(self.schema, self.table)
+
+    def complete(self):
+        if not self.executed:
+            return False
+        return self.output().exists()
```

Comment:
The destination table always exists (the task is overwriting it). If we just check it the table exists the task would never execute

Reply:
Uhm, I did not notice that. Wouldn't that be the case that each time that we run the parent task we would get a more simplified version each time? I mean, first time you simplify, next time, this will run again (because of the not executed part) and it would simplify the already simplified table, leading to even more simplified geos. Wouldn't it?